pkg/apis/core/v1beta1/helper/condition_builder.go

```diff
@@ -104,10 +118,15 @@ func (b *defaultConditionBuilder) WithNowFunc(now func() metav1.Time) ConditionB
 // - The error codes will not be transferred from the old to the new condition
 func (b *defaultConditionBuilder) Build() (new gardencorev1beta1.Condition, updated bool) {
 	var (
-		now       = b.nowFunc()
+		now       = metav1.Time{Time: b.clock.Now()}
 		emptyTime = metav1.Time{}
 	)
 
+	// TODO(oliver-goetz): Delete when removing `WithNowFunc`
+	if b.nowFunc != nil {
+		now = b.nowFunc()
+	}
```

Comment:
Maybe this is no longer needed after above suggestion?

Reply:
I removed `WithNowFunc` entirely now.